### PR TITLE
misc(v2): better dx: yarn clear, yarn watch, yarn serve

### PR DESCRIPTION
**Pull request by slorber**
- `yarn clean`: cleanup of useful folders (ts incremental build, babel cache, /lib...)

- `yarn watch`: start incremental TS compiltation in --watch mode over the whole project

- `yarn serve`: use `serve` to serve a production build of the project 

I tried to create a `watchPlugin` on the website, because I don't like constantly restarting the site, but I don't think it's possible to live-update the plugins' code at runtime :(

---

The watch mode, even without hot code reload, could be useful. We could watch all packages, and only need to run `yarn start:v2` to restart the site, instead of `yarn start` that will recompile each package. The developer feedback would be faster.

### CODE REVIEW

**Comment by slorber** [comment]
@yangshun @lex111 what do you think of including `docusaurus clear` and `docusaurus serve` commands? 

I think it could be useful for other users as well.

File: package.json [hunk 1]
```diff
@@ -22,7 +25,9 @@
     "lerna": "lerna",
     "test": "jest",
     "test:build:v2": "./admin/scripts/test-release.sh",
-    "tsc": "lerna run tsc --no-private"
+    "tsc": "lerna run tsc --no-private",
+    "watch": "yarn lerna run --parallel --no-private watch",
```
Comment (slorber) [inline package.json:29]: start all TS subprojects in watch mode in parallel

File: packages/docusaurus/package.json [hunk 2]
```diff
@@ -23,7 +23,8 @@
     "docusaurus": "bin/docusaurus.js"
   },
   "scripts": {
-    "tsc": "tsc && tsc -p tsconfig.client.json && node copyUntypedFiles.js"
+    "tsc": "tsc && tsc -p tsconfig.client.json && node copyUntypedFiles.js",
+    "watch": "node copyUntypedFiles.js && concurrently -n \"server,client\" --kill-others \"tsc --watch\" \"tsc -p tsconfig.client.json --watch\""
```
Comment (slorber) [inline packages/docusaurus/package.json:27]: concurrently needed because we need to compile for server+client here

File: website/watchPlugin.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+  );
+
+  console.log('TSPathsToWatch', TSPathsToWatch);
+
+  return {
+    name: 'docusaurus-website-watch-plugin',
+    getPathsToWatch() {
+      return [...TSPathsToWatch];
+    },
+  };
```
Comment (slorber) [inline website/watchPlugin.js:30]: Unless you confirm it's possible, I'm going to remove this, as I don't think it's possible to reload at runtime plugin code

File: website/watchPlugin.js [hunk 4]
```diff
@@ -0,0 +1,31 @@
+const path = require('path');
```
Comment (yangshun) [inline website/watchPlugin.js:1]: Let's put this under `website/plugins`?

Also, we'd need copyright headers.

File: website/watchPlugin.js [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    if (!fs.existsSync(tsPackageFolder)) {
+      throw new Error(`TS package does not exist: ${tsPackageFolder}`);
+    }
+  });
+
+  const TSPathsToWatch = TSPackageFolders.map(
+    (tsPackagePath) => `${tsPackagePath}/lib/**/*.*`,
+  );
+
+  console.log('TSPathsToWatch', TSPathsToWatch);
```
Comment (yangshun) [inline website/watchPlugin.js:23]: Remove before merging?

File: website/watchPlugin.js (hunk 3 above)
Comment (yangshun) [reply website/watchPlugin.js:30]: Nope it's not possible now. I have to stop the development server and restart it. Hmm maybe this can be an MLH project?

**Comment by slorber** [comment]
@yangshun I think it's complicated to hot reload the D2 code, tried to hack a bit with nodejs child processes / spawn etc but it's not easy.

Found a solution to automatically restart the server. It's probably not the fastest DX we could get as each restart takes a few seconds, but it's still better than current DX and is a tiny change so could do the job for now:

```
    "start:v2:watch": "nodemon --watch \"./packages/*/lib/**/*.*\" --exec \"yarn start:v2\"",
```

**Pull request merged**